doc/oomd/SONiC OOM daemon.md

```diff
@@ -0,0 +1,255 @@
+| systemd-oomd.service | Require cgroupv2, not scalable, can't customize to only termi. | [systemd-oomd.service(8) - Linux manual page (man7.org)](https://man7.org/linux/man-pages/man8/systemd-oomd.service.8.html) |
+
+
+
+## 3.2 Default OOMD config
+- OOMD will be enabled by default.
+- Default high-water mark: 90
+- Default low-water mark: 60
+- Default user/group list is empty.
+- Terminate domain user process by default.
+
+## 3.3 ConfigDB Schema
+ - OOMD setting table.
+```
+; Key
+key                      = "policies"         ; OOMD policies
+; Attributes
+enable                   = Boolean            ; Enable status, true for enable.
+high_water_mark          = 3*DIGIT            ; Memory utilzation high-water mark
+low_water_mark           = 3*DIGIT            ; Memory utilzation low-water mark
+domain_account           = Boolean            ; OOMD will terminate process run by domain account
+terminate_scope          = LIST(1*32VCHAR)    ; OOMD terminate scope: "process" or "session"
+allow_user_list          = LIST(string)    ; OOMD can terminate process start by users in this list 
+allow_group_list         = LIST(string)    ; OOMD can terminate process start by group users in this list 
+```
+ - Yang model: [sonic-system-oomd.yang](./sonic-system-oomd.yang)
+
+## 3.4 CLI
+ - Add following command to change OOMD setting.
+```
+    // config OOMD enable/disable status
+    config oomd { enable/disable}
+    
+    // config OOMD high-water mark and low-water mark, number are memory utilzation percentage.
+    config oomd policy { highwatermark|lowwatermark } <number>
+    
+    // config which user/group can be terminate
+    config oomd policy { users|groups } { add|del } <name>
+    
+    // config if OOMD can terminate domain account process/session
+    config oomd policy domainaccount { enable|disable }
+    
+    // config OOMD terminate user session or process
+    config oomd policy terminatescope { session|process }
+    
+    // config OOMD log level
+    config oomd policy loglevel { debug|standard }
+
+```
+
+ - Add following command to show OOMD setting.
+```
+    // show oomd setting
+    show oomd policy
+
+    // show oomd enable/disable status
+    show oomd status
+```
+
+# 4 Error handling
+ - OOMD will write errors to syslog.
+
+# 5 Serviceability and Debug
+ - OOMD  can be debugged by enabling the debug flag OOMD config.
+
+# 6 Unit Test
+
+## 6.1 Enable/disable OOMD test
+
+  - Enable/disable OOMD and check the OOMD service status:
+  ```
+      Verify the OOMD service started when OOMD enabled.
+      Verify the OOMD service stopped when OOMD disabled.
+  ```
+
+## 6.2 OOMD policy test
+
+  - Change OOMD high-water/low-water mark and check the OOMD will be triggered correctly:
+  ```
+      Verify when system memory utilzation reach high-water mark OOMD triggered.
+      Verify when OOMD triggered, procress been terminated to free enough memory to low-water mark.
```

Comment:
>procress [](http://example.com/codeflow?start=34&length=8)

process? many times in this markdown.

Reply:
Fixed, typo.

Reply:
I see you fixed some. For this one, you deleted it and then there is no subject.

Reply:
Fixed, add back.